src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAWindowsDesktopProject.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildAWindowsDesktopProject : SdkTest
+    {
+        public GivenThatWeWantToBuildAWindowsDesktopProject(ITestOutputHelper log) : base(log)
+        {}
+
+        [WindowsOnlyRequiresMSBuildVersionTheory("16.7.0-preview-20310-07")]
+        [InlineData("UseWindowsForms")]
+        [InlineData("UseWPF")]
+        public void It_errors_when_missing_windows_target_platform(string propertyName)
+        {
+            var targetFramework = "net5.0";
+            TestProject testProject = new TestProject()
+            {
+                Name = "MissingTargetPlatform",
+                IsSdkProject = true,
+                TargetFrameworks = targetFramework
+            };
+            testProject.AdditionalProperties[propertyName] = "true";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
```

Comment:
Pass `propertyName` as the optional identifier parameter to `CreateTestProject` so that the different instances of the test will use different project folders.